tests/models/test_utils.py

```diff
@@ -0,0 +1,135 @@
+from collections import namedtuple
+
+import pytest
+from sklearn import datasets
+import sklearn.neighbors as knn
+import mlflow
+import random
+
+from mlflow.models import add_libraries_to_model
+from mlflow.models.utils import get_model_version_from_model_uri
+
+ModelWithData = namedtuple("ModelWithData", ["model", "inference_data"])
+
+
+@pytest.fixture(scope="module")
+def sklearn_knn_model():
+    iris = datasets.load_iris()
+    X = iris.data[:, :2]  # we only take the first two features.
+    y = iris.target
+    knn_model = knn.KNeighborsClassifier()
+    knn_model.fit(X, y)
+    return ModelWithData(model=knn_model, inference_data=X)
+
+
+def random_int(lo=1, hi=1000000000):
+    return random.randint(lo, hi)
+
+
+def test_adding_libraries_to_model_default(sklearn_knn_model):
```

Comment:
Can we test the behavior when `model_uri` refers to a model version that does not have a source run? (A new run should be created)